### PR TITLE
add(bot): github enterprise support

### LINKED ISSUE #562: [self-hosted] 500 error on status_event
Hey there, looking forward to utilizing this app! We're trying to get it set up on our own architecture but are running into this error on status events:

```
INFO uvicorn:httptools_impl.py:436 (REDACTED, REDACTED) - "POST /api/github/hook HTTP/1.1" 500
ERROR uvicorn:httptools_impl.py:371 Exception in ASGI application
Traceback (most recent call last):
  File "/var/app/.venv/lib/python3.7/site-packages/uvicorn/protocols/http/httptools_impl.py", line 368, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/var/app/.venv/lib/python3.7/site-packages/fastapi/applications.py", line 140, in __call__
    await super().__call__(scope, receive, send)
  File "/var/app/.venv/lib/python3.7/site-packages/starlette/applications.py", line 134, in __call__
    await self.error_middleware(scope, receive, send)
  File "/var/app/.venv/lib/python3.7/site-packages/starlette/middleware/errors.py", line 178, in __call__
    raise exc from None
  File "/var/app/.venv/lib/python3.7/site-packages/starlette/middleware/errors.py", line 156, in __call__
    await self.app(scope, receive, _send)
  File "/var/app/.venv/lib/python3.7/site-packages/sentry_sdk/integrations/asgi.py", line 51, in _run_app
    raise exc from None
  File "/var/app/.venv/lib/python3.7/site-packages/sentry_sdk/integrations/asgi.py", line 48, in _run_app
    await callback()
  File "/var/app/.venv/lib/python3.7/site-packages/starlette/exceptions.py", line 73, in __call__
    raise exc from None
  File "/var/app/.venv/lib/python3.7/site-packages/starlette/exceptions.py", line 62, in __call__
    await self.app(scope, receive, sender)
  File "/var/app/.venv/lib/python3.7/site-packages/starlette/routing.py", line 590, in __call__
    await route(scope, receive, send)
  File "/var/app/.venv/lib/python3.7/site-packages/starlette/routing.py", line 208, in __call__
    await self.app(scope, receive, send)
  File "/var/app/.venv/lib/python3.7/site-packages/starlette/routing.py", line 41, in app
    response = await func(request)
  File "/var/app/.venv/lib/python3.7/site-packages/fastapi/routing.py", line 133, in app
    raw_response = await dependant.call(**values)
  File "./kodiak/main.py", line 99, in webhook_event
    await handle_webhook_event(event_name=github_event, payload=event)
  File "./kodiak/event_handlers.py", line 227, in handle_webhook_event
    await push(PushEvent.parse_obj(payload))
  File "./kodiak/event_handlers.py", line 166, in push
    prs = await api_client.get_open_pull_requests(base=branch_name)
  File "./kodiak/queries.py", line 985, in get_open_pull_requests
    headers = await get_headers(installation_id=self.installation_id)
  File "./kodiak/queries.py", line 1190, in get_headers
    token = await get_token_for_install(installation_id=installation_id)
  File "./kodiak/queries.py", line 1183, in get_token_for_install
    assert res.status_code < 300
AssertionError
```

Events such as `requested` and `synchronize` seem to be working (200 response) but any status check events are throwing this error. I'm looking for information on understanding this error and seeing if we can get it resolved. Please let me know if anything else is required from me to help debug this issue.

### PR DESCRIPTION
**Pull request by rdmulford**
resolves #562 

adds new configuration options to allow setting the github root and base paths and add additional request headers to git api calls which allows kodiak to be used with self-hosted github instances. Our instance also had a different path to our graphql endpoint so i made that a seperate env.

### CODE REVIEW

**Review by chdsbd — COMMENTED** [review]
Thanks for the PR, it looks great.

I was super surprised that the GitHub Enterprise API paths aren't the same as GitHub.com.

I think doing a null check on the API headers should fix the bot_test failures. the build_container jobs are okay to fail because this PR is from a fork.

File: bot/kodiak/queries.py [hunk 1]
```diff
@@ -742,6 +742,7 @@ def __init__(self, *, owner: str, repo: str, installation_id: str):
         self.session.headers[
             "Accept"
         ] = "application/vnd.github.antiope-preview+json,application/vnd.github.merge-info-preview+json"
+        self.session.headers[conf.GITHUB_API_HEADER_NAME] = conf.GITHUB_API_HEADER_VALUE
```
[the remark this answers is not in the document]
Comment (rdmulford) [reply bot/kodiak/queries.py]: yep that makes sense good call!

File: bot/kodiak/app_config.py [hunk 2]
```diff
@@ -35,6 +35,19 @@
...
 SUBSCRIPTIONS_ENABLED = config("SUBSCRIPTIONS_ENABLED", cast=bool, default=False)
 
+# change the API root for self-hosted GitHub instances
+GITHUB_API_ROOT = config("GITHUB_API_ROOT", default="https://api.github.com")
+
+# additional path to github api
+GITHUB_API_PATH = config("GITHUB_API_PATH", default=None)
+
+# additional path to github graphql endpoint
+GITHUB_API_GRAPHQL_PATH = config("GITHUB_API_GRAPHQL_PATH", default=None)
```
[the remark this answers is not in the document]
Comment (rdmulford) [reply bot/kodiak/app_config.py]: I like that refactor, ive gone ahead and updated to use these two options 👍

**Comment by chdsbd** [comment]
I tested this change and it looks good to merge.

**Review by sbdchd — APPROVED** [review]

**Pull request merged**